packages/yarnpkg-core/sources/miscUtils.ts

```diff
@@ -232,38 +232,63 @@ export class DefaultStream extends Transform {
 // Webpack has this annoying tendency to replace dynamic requires by a stub
 // code that simply throws when called. It's all fine and dandy in the context
 // of a web application, but is quite annoying when working with Node projects!
+declare const __non_webpack_require__: typeof require | undefined;
 
-export function dynamicRequire(path: string) {
-  // @ts-expect-error
-  if (typeof __non_webpack_require__ !== `undefined`) {
-    // @ts-expect-error
-    return __non_webpack_require__(path);
-  } else {
-    return require(path);
-  }
-}
+export let dynamicRequire: (path: string) => any;
 
-export function dynamicRequireNoCache(path: PortablePath) {
-  const physicalPath = npath.fromPortablePath(path);
+if (typeof __non_webpack_require__ !== `undefined`)
+  dynamicRequire = path => __non_webpack_require__(path);
+else
+  dynamicRequire = path => require(path);
 
-  const currentCacheEntry = require.cache[physicalPath];
-  delete require.cache[physicalPath];
+export let dynamicRequireNoCache: (path: PortablePath) => any;
 
-  let result;
-  try {
-    result = dynamicRequire(physicalPath);
+if (typeof __non_webpack_require__ !== `undefined`) {
+  dynamicRequireNoCache = path => {
+    const physicalPath = npath.fromPortablePath(path);
+
+    const currentCacheEntry = __non_webpack_require__.cache[physicalPath];
+    delete __non_webpack_require__.cache[physicalPath];
 
-    const freshCacheEntry = require.cache[physicalPath];
-    const freshCacheIndex = module.children.indexOf(freshCacheEntry);
+    let result;
+    try {
+      result = __non_webpack_require__(physicalPath);
 
-    if (freshCacheIndex !== -1) {
-      module.children.splice(freshCacheIndex, 1);
+      const freshCacheEntry = __non_webpack_require__.cache[physicalPath];
+      const freshCacheIndex = module.children.indexOf(freshCacheEntry);
```

Comment:
This is the webpack module, not the node module. This doesn't really do anything in this variant of the function.
Webpack doesn't have a non-webpack module variable analogous to `__non_webpack_require__`, so there's no easy fix for this.

I'm not sure what this is supposed to do, do we need this?

Reply:
You can use `eval('module')` to bypass Webpack and get the true module - would that be enough? Also, do we need two implementations? Can't we use a single one with different variable bindings?

Reply:
> You can use eval('module') to bypass Webpack and get the true module - would that be enough?

That should do!

> Also, do we need two implementations? Can't we use a single one with different variable bindings?

In webpack the `require.cache` is actually `__webpack_require__.c`, so I duplicated things. I now realise that this was indeed unnecessary, as we never use `__webpack_require__` in this context.